app/browser/reducers/spellCheckerReducer.js

```diff
@@ -28,12 +31,29 @@ const migrate = (state) => {
   return state
 }
 
+const setSpellCheckerSettings = () => {
+  const enabled = getSetting(settings.SPELLCHECK_ENABLED)
+  setUserPref('browser.enable_spellchecking', enabled)
+  if (enabled) {
+    setUserPref('spellcheck.dictionaries',
+                getSetting(settings.SPELLCHECK_LANGUAGES))
+  }
+}
+
 const spellCheckerReducer = (state, action, immutableAction) => {
   action = immutableAction || makeImmutable(action)
   switch (action.get('actionType')) {
     case appConstants.APP_SET_STATE:
       state = migrate(state)
       break
+    case appConstants.APP_CHANGE_SETTING:
+      // TODO(darkdh): APP_SET_STATE is too early for startup setting so we use
```

Comment:
why is it too early?

Reply:
I don't know the root cause but when I tried to use APP_SET_STATE, it has no effect for these two preferences.